### PR TITLE
[EWT-63] `scheme_selection` support for single payments, smart retries for recurring payments

### PR DESCRIPTION
**Pull request by dili91**
# Description

- [x] Adds support for the `scheme_selection`
- [x] includes a couple of refactor which I had forgotten to include in one of my recent PR 😞 
- [x] Adds support for Smart retries on recurring payments

## Type of change

Please select multiple options if required.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the relevant documentation

### CODE REVIEW

**Comment by dili91** [comment]
As we have to bump the major version I'm happy to include in this PR also the contents of REC-656

File: src/test/java/com/truelayer/java/integration/PaymentsIntegrationTests.java [hunk 1]
```diff
@@ -85,11 +92,16 @@ public void shouldReturnErrorIfSignatureIsInvalid() {
     }
 
     @DisplayName("It should get the payment details")
-    @ParameterizedTest(name = "of a payment with status {0}")
-    @ValueSource(strings = {"AUTHORIZATION_REQUIRED", "AUTHORIZING", "AUTHORIZED", "EXECUTED", "SETTLED", "FAILED"})
+    @ParameterizedTest(name = "of a payment of type {0} with status {1}")
+    @MethodSource("provideShouldReturnAPaymentDetailTestParameters")
```
Comment (dili91) [inline src/test/java/com/truelayer/java/integration/PaymentsIntegrationTests.java:96]: this also improves the output of our tests

File: src/main/java/com/truelayer/java/payments/entities/retry/SmartRetry.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.truelayer.java.payments.entities.retry;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+
+@Getter
+@Builder
+@ToString
```
Comment (dili91) [inline src/main/java/com/truelayer/java/payments/entities/retry/SmartRetry.java:10]: This `@ToString` is use to improve test readability :
Screenshot 2022-12-14 at 18 49 34

File: README.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ TrueLayerClient client = TrueLayerClient.New()
         .build();
 ```
 
-More details about our HTTP logs can be found [here](https://github.com/TrueLayer/truelayer-java/wiki/HTTP-logging).
+More details about our logs can be found [ong our wiki](https://github.com/TrueLayer/truelayer-java/wiki/Logging).
```
Comment (tl-luca-baggi) [inline README.md:90]:
```suggestion
More details about our logs can be found [on our wiki](https://github.com/TrueLayer/truelayer-java/wiki/Logging).
```

**Review by tl-luca-baggi — APPROVED** [review]

**Pull request merged**